Parsers/ASimDns/Parsers/vimDnsFortinetFortigate.yaml

```diff
@@ -0,0 +1,273 @@
+          (array_length(domain_has_any) == 0 or AdditionalExtensions has_any (domain_has_any)) and
+          (responsecodename == '*' or AdditionalExtensions has DnsResponseCodeNameLookup) and
+          (response_has_ipv4 == '*' or has_ipv4(AdditionalExtensions, response_has_ipv4)) and 
+          (array_length(response_has_any_prefix) == 0 or has_any_ipv4_prefix(AdditionalExtensions, response_has_any_prefix)) and
+          (eventtype == '*' or eventtype == "Query")
+      | project TimeGenerated, EventOriginalSubType = DeviceEventClassID, AdditionalExtensions, EventUid = _ItemId, EventOriginalSeverity = LogSeverity, EventProductVersion = DeviceVersion ,Computer, Type, SrcIpAddr = SourceIP, SrcPortNumber = SourcePort, DstIpAddr = DestinationIP, DstPortNumber = DestinationPort, EventMessage = Message, NetworkProtocolNumber = Protocol
+      | lookup DeviceEventClassIDLookup on EventOriginalSubType
+      | parse-kv  AdditionalExtensions as (FTNTFGTlogid:string, FTNTFGTsubtype:string, FTNTFGTsrccountry:string, FTNTFGTdstcountry:string,FTNTFGTsrcintfrole:string, FTNTFGTrcode:string, FTNTFGTqname:string, FTNTFGTqtype:string, FTNTFGTxid:string, FTNTFGTqtypeval:int, FTNTFGTqclass:string, FTNTFGTcatdesc:string, FTNTFGTipaddr:string, FTNTFGTunauthuser:string, FTNTFGTuser:string) with (pair_delimiter=";", kv_delimiter="=")
+      | where (array_length(domain_has_any) == 0 or FTNTFGTqname has_any (domain_has_any)) and
+          (responsecodename == '*' or FTNTFGTrcode == DnsResponseCodeNameLookup) and
+          (response_has_ipv4 == '*' or has_ipv4(FTNTFGTipaddr, response_has_ipv4)) and
+          (array_length(response_has_any_prefix) == 0 or has_any_ipv4_prefix(FTNTFGTipaddr, response_has_any_prefix))
+      | project-rename 
+          EventOriginalResultDetails  = FTNTFGTrcode,
+          EventOriginalUid            = FTNTFGTlogid,
+          DvcZone                     = FTNTFGTsrcintfrole,
+          EventOriginalType           = FTNTFGTsubtype,
+          SrcGeoCountry               = FTNTFGTsrccountry,
+          DstGeoCountry               = FTNTFGTdstcountry,
+          DnsQuery                    = FTNTFGTqname,
+          DnsQueryTypeName            = FTNTFGTqtype,
+          TransactionIdHex            = FTNTFGTxid,
+          DnsQueryClass               = FTNTFGTqtypeval,
+          DnsQueryClassName           = FTNTFGTqclass,
+          UrlCategory                 = FTNTFGTcatdesc,
+          DnsResponseName             = FTNTFGTipaddr
+      | extend 
+          DnsQueryTypeName = case(
+              DnsQueryTypeName == "Unknown","",
+              DnsQueryTypeName
+          )
+      | lookup EventOriginalResultDetailsLookup on EventOriginalResultDetails
+      | lookup DnsQueryTypeLookup on DnsQueryTypeName
+      | invoke _ASIM_ResolveDvcFQDN ("Computer")
+      | invoke _ASIM_ResolveNetworkProtocol("NetworkProtocolNumber")
+      | extend 
+          SrcUsername         = coalesce(FTNTFGTuser, FTNTFGTunauthuser),
```

Comment:
we have user info coming in a different field, can you please check? Also we have sessionid info here in this sample log:
https://docs.fortinet.com/document/fortigate/7.4.1/fortios-log-message-reference/216594/dns-log-support-for-cef

Reply:
Added sessionid

Reply:
@KustoKing 
did you find any sample where sessionid was present?
I'll try validating these field from my end too next week. Currently, going on vacation.

Reply:
Changed sessionid to ExtID